tests/src/core/users/DetermineLifetimeConditions.test.ts

```diff
@@ -0,0 +1,214 @@
+
+      jest.spyOn(paymentService, 'getCustomer').mockResolvedValue(customer as Stripe.Response<Stripe.Customer>);
+      jest.spyOn(paymentService, 'getCustomersByEmail').mockResolvedValue([customer]);
+      jest.spyOn(paymentService, 'getInvoicesFromUser').mockResolvedValue([]);
+      jest.spyOn(tiersService, 'getTiersProductsByUserId').mockRejectedValue(tierNotFoundError);
+      jest.spyOn(determineLifetimeConditions as any, 'getHigherTier').mockResolvedValue(null);
+
+      await expect((determineLifetimeConditions as any).handleStackingLifetime(user)).rejects.toThrow(
+        tierNotFoundError,
+      );
+    });
+
+    it('When we want to fetch the higher tier and the max space bytes, then the correct tier and bytes are returned', async () => {
+      const user = getUser({ lifetime: true });
+      const customer = getCustomer({ id: user.customerId });
+      const invoice = getInvoice();
+      const mockedTier = newTier({ billingType: 'lifetime' });
+
+      jest.spyOn(paymentService, 'getCustomer').mockResolvedValue(customer as Stripe.Response<Stripe.Customer>);
+      jest.spyOn(paymentService, 'getCustomersByEmail').mockResolvedValue([customer]);
+      jest.spyOn(paymentService, 'getInvoicesFromUser').mockResolvedValue([invoice]);
+      jest.spyOn(determineLifetimeConditions as any, 'getPaidInvoices').mockResolvedValue([invoice]);
+      jest.spyOn(tiersService, 'getTiersProductsByUserId').mockResolvedValue([mockedTier]);
+      jest.spyOn(determineLifetimeConditions as any, 'getHigherTier').mockResolvedValue(mockedTier);
+
+      const result = await (determineLifetimeConditions as any).handleStackingLifetime(user);
+
+      expect(result.tier).toEqual(mockedTier);
+      expect(result.maxSpaceBytes).toBe(parseInt(invoice.lines.data[0].price?.metadata?.maxSpaceBytes ?? '0'));
+    });
+  });
+
+  describe('Get paid invoices', () => {
+    it('When there is no metadata in the invoice, then the invoice should be skipped', async () => {
+      const customer = getCustomer();
+      const invoice = getInvoice();
+      invoice.lines.data[0].price!.metadata = {};
+
+      const result = await (determineLifetimeConditions as any).getPaidInvoices(customer, [invoice]);
+
+      expect(result).toEqual([]);
+    });
+
+    it('When the invoice is paid out of band, then the invoice is returned directly', async () => {
+      const customer = getCustomer();
+      const invoice = getInvoice();
+      invoice.paid = true;
+      invoice.paid_out_of_band = true;
+      invoice.lines.data[0].price!.metadata!.planType = 'one_time';
+      invoice.metadata = {};
+
+      const result = await (determineLifetimeConditions as any).getPaidInvoices(customer, [invoice]);
+
+      expect(result).toEqual([invoice]);
+    });
+
+    it('When the invoice is paid and it has not been refunded nor disputed, then the invoice is returned', async () => {
```

Comment:
This case may be providing a false positive as the if will check one of those two status before and will not check the other as the code base uses an AND. Use one test case for refunded and other for disputed